versions.tf

```diff
@@ -1,7 +1,7 @@
 terraform {
-  required_version = ">= 0.15"
+  required_version = ">= 1.2.1"
 
   required_providers {
-    azurerm = ">=2.91"
+    azurerm = ">=3.10.0"
```

Comment:
Can we make this `>=3.0` instead (unless there's a specific reason the 3.0-3.9 versions aren't acceptable). Not really an "important" distinction but it's closer to the principle of allowing the oldest version possible.